File: NHSOnline.Backend.Worker.IntegrationTests/Mocking/Models/Mapping.cs
namespace NHSOnline.Backend.Worker.IntegrationTests.Mocking.Models
{
    public class Mapping
    {
        public Mapping(Request request, Response response)
        {
            Request = request;
            Response = response;
        }

        public Mapping(Request request, Response response, int priority)
        {
            Request = request;
            Response = response;
            Priority = priority;
        }

        public int Priority { get; set; }
        public Request Request { get; set; }
        public Response Response { get; set; }
    }
}

File: src/main/kotlin/mocking/models/Mapping.kt
package mocking.models

import com.google.gson.Gson
import mocking.MockingClient
import java.time.Duration

class Mapping() {

    var request: Request? = null
    var response: Response? = null


    constructor(request: Request, response: Response) : this() {
        this.request = request
        this.response = response
    }

    fun delayedBy(seconds: Duration): Mapping {
        this.response!!.fixedDelayMilliseconds = seconds.toMillis().toInt()
        return this
    }

    override fun toString(): String {
        val requestString = "Request: ${Gson().toJson(request)}"
        val responseString = "Response: ${Gson().toJson(response)}"
        return "Mapping: $requestString, $responseString"
    }
}